Arduino nano RP2040 and portenta H7 samples

RLeclair opened the pull request:
Samples for Arduino Nano RP2040 and Portenta H7. Includes a sample for hub and central for each board. Adapted from the following PRs: 

#23 , #22 , #19 , #18

Abandoned the above PRs to consolidate in a single PR and addressed the comments left on this PR.

RLeclair committed:
Merging nano and portenta samples

RLeclair committed:
Addressing comments

RLeclair committed:
Addressing comments Central Arduino Nano

RLeclair committed:
Addressing comments Central Arduino Portenta

RLeclair committed:
Adding support for ECCX08 with hmac update

RLeclair committed:
Renaming and removing mbedtls includes

RLeclair committed:
Adding install instructions for ArduinoECCX08

danewalton-msft commented:
Since this is a large PR, are you able to call out what you know is different about these samples compared to the ones that already exist? Most of these look like `CTRL-C | CTRL-V` code from other samples we have.

RLeclair commented:
> Since this is a large PR, are you able to call out what you know is different about these samples compared to the ones that already exist? Most of these look like `CTRL-C | CTRL-V` code from other samples we have.

Yes, the changes I made are:
- Removed the need for mbed tls to perform SAS token signing. Using the Arduino ECCX08 library to leverage the ATECC chip on board. 
- Changed instructions to reflect the need to install ECCX08.
- From comments on the previous PRs:
   - Moved `Wifi.begin` outside of while loop in all samples.
   - Reduced delay in main loop of all samples.
   - Added initialization of `telemetryNextSendTimeMs` in setup of hub samples.
   - Added exit loop macro for hub samples. 
   - Changed the logs to provide uniformity. Used the Serial Logger model used in hub ESP32 samples.
       - Note that for the Central samples I did not use the Serial Logger. Central samples are dependent on AzureIot.h for functionality and it already includes logging functions.
   - Minor changes in variable naming for clarity.

danewalton-msft commented:
@RLeclair awesome thanks!

danewalton-msft commented on examples/Azure_IoT_Central_Arduino_Nano_RP2040_Connect/Azure_IoT_PnP_Template.h:
```diff
@@ -0,0 +1,104 @@
+
+#include "AzureIoT.h"
+
+/*
```
nit: not sure if this was done on purpose or not, but just for future reference (since this won't actually be generated doxygen docs). Doxygen comments start with two stars to separate them from general, multiline comments:
```suggestion
/**
```

danewalton-msft replied:
don't actually have to make this change.

RLeclair replied:
Good to know, thanks for pointing it out!

danewalton-msft commented on examples/Azure_IoT_Central_Arduino_Nano_RP2040_Connect/iot_configs.h:
```diff
@@ -0,0 +1,24 @@
+
+// User-agent (url-encoded) provided by the MQTT client to Azure IoT Services.
+// When developing for your own Arduino-based platform, please follow the format '(ard;<platform>)'.
+#define IOT_CONFIG_AZURE_SDK_CLIENT_USER_AGENT "c%2F" AZ_SDK_VERSION_STRING "(ard;portentaH7)"
```
```suggestion
#define IOT_CONFIG_AZURE_SDK_CLIENT_USER_AGENT "c%2F" AZ_SDK_VERSION_STRING "(ard;rp2040)"
```

danewalton-msft replied:
might just want to double check all of these

danewalton-msft commented on examples/Azure_IoT_Central_Arduino_Portenta_H7/Azure_IoT_PnP_Template.cpp:
```diff
@@ -0,0 +1,546 @@
+#include <az_precondition_internal.h>
+
+/* --- Defines --- */
+#define AZURE_PNP_MODEL_ID "dtmi:azureiot:devkit:freertos:Esp32AzureIotKit;1"
```
double checking that we're sticking with this since it has similar sensors?

RLeclair replied:
Resolved, template is for testing purposes

danewalton-msft commented on examples/Azure_IoT_Central_Arduino_Nano_RP2040_Connect/Azure_IoT_PnP_Template.cpp:
```diff
@@ -0,0 +1,544 @@
+#define SAMPLE_TOTAL_MEMORY_PROPERTY_NAME              "totalMemory"
+
+#define SAMPLE_MANUFACTURER_PROPERTY_VALUE             "Arduino"
+#define SAMPLE_MODEL_PROPERTY_VALUE                    "Arduino Nano RP2040 Connet"
```
Do a find all and replace on `Connet` just to double check
```suggestion
#define SAMPLE_MODEL_PROPERTY_VALUE                    "Arduino Nano RP2040 Connect"
```

RLeclair committed:
Addressing comments

momuno commented:
Thank you @RLeclair for all your work on this!
